### PR TITLE
Milestone 4: Add calculator logic for performing calculations.

### PR DESCRIPTION
**Pull request by Abhigyan001**
- Added logic folder and include calculate.js and operate.js
- Created operate function and basic arithmetic operation in operate.js
- Imported the calculate function to the App component

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hello @Abhigyan001  👋

**Good job here. You did well using a try/catch block on big.js operations 👍**

However, let's make just a few changes and we are all set.

- Please refer to comments on the code.

### **STATUS: CHANGES REQUESTED** 👷‍♂️

Kindly fix these issues and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

HAPPY CODING👨‍💻

File: src/logic/calculate.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  }
+
+  if (['+', 'X', '-', '÷', '='].includes(buttonName)) {
+    if (total && next && operation) {
+      const newOperation = buttonName === '=' ? '' : buttonName;
+      dataResult.total = operate(total, next, operation);
+      dataResult.operation = newOperation;
+    }
+
+    if (total && !next) {
```
Comment (mikenath223) [inline src/logic/calculate.js:38]: - [ ] OPTIONAL - Consider using a switch statement here. Its much more concise.

File: src/logic/operate.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      result = Big(one.times(two));
+    }
+
+    if (operation === '÷') {
+      result = Big(one.div(two));
+    }
+
+    if (operation === '%') {
+      result = Big(one.div(two));
+    }
```
Comment (mikenath223) [inline src/logic/operate.js]: - [ ] This should be changed to division by `100`. It's a percent operator.

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hello again @Abhigyan001 👋

**You've done pretty well. Good job working on the optional change. 👍**

- However, just one more fix. Just hang in there. 

### **STATUS: CHANGES REQUESTED** 👷‍♂️

Kindly fix this issue and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

HAPPY CODING👨‍💻

File: src/logic/operate.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        result = n1.times(n2);
+        break;
+
+      case '÷':
+        result = n1.div(n2);
+        break;
+
+      case '%':
+        result = n1.mod(n2);
+        break;
```
Comment (mikenath223) [inline src/logic/operate.js]: - [ ] As stated in the previous review this is a percentage operator and, not a modulus operator. So kindly divide by `100` instead.

**Review by MuriloRoque — APPROVED** [review]
## Status: Approved :green_circle: 

### Hello @Abhigyan001 :wave: 

### Thank you for making all of the requested changes. Your project is approved :clap: :clap: :clap: 

### You may merge, good luck with your next projects :rocket: 

Murilo Roque Paiva da Silva

- Github: [@muriloroque](https://github.com/MuriloRoque)
- Linkedin: [MuriloRoque](https://www.linkedin.com/in/murilo-roque-b1268741/)

[Twitter Follow](https://twitter.com/MuriloRoquePai1)

**Pull request merged**